File: mall/src/main/java/com/zxy/mall/http/ServiceInterface.java
package com.zxy.mall.http;

import com.zxy.frame.net.BaseResponse;

import io.reactivex.Observable;
import okhttp3.ResponseBody;
import retrofit2.Call;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.GET;
import retrofit2.http.Header;
import retrofit2.http.POST;
import retrofit2.http.Streaming;
import retrofit2.http.Url;

/**
 * Created by zxy on 2019/08/06.
 */
public interface ServiceInterface {

    String BASE_URL = "http://api.dev.2048.com";
    String GET_TAB = "/api/get_tab";
    String SQUARE_POST = "/api/square_post";
    String VERSION_CHECK = "/api/version_check";

    @Streaming
    @GET
    Observable<ResponseBody> downloadFile(@Header("RANGE") String downParam, @Url String fileUrl);

//    @POST(GET_TAB)
//    Call<ResponseBody> get_tab();
//
//    /**
//     * 社区广场列表
//     *
//     * @param page
//     * @return
//     */
//    @FormUrlEncoded
//    @POST(SQUARE_POST)
//    Observable<BaseResponse<SquarePostEntity>> square_post(@Field("page") int page);
//
//    /**
//     * 版本更新
//     *
//     * @return
//     */
//    @POST(VERSION_CHECK)
//    Observable<BaseResponse<VersionUpdateEntity>> version_check();

}

File: app/src/main/java/com/zxy/demo/httpbin/ServiceInterface.kt
package com.zxy.demo.httpbin

import okhttp3.RequestBody
import okhttp3.ResponseBody
import retrofit2.Call
import retrofit2.http.*
import java.io.File

/**
 * Created by Admin
 * on 2020/8/13
 */
interface ServiceInterface {

    companion object {
        var BASE_URL = "http://www.httpbin.org/"
    }

    @Streaming
    @GET
    fun downloadFile(
        @Header("RANGE") downParam: String,
        @Url fileUrl: String
    ): Call<ResponseBody>

    @GET("/get")
    fun get(@Query("name")name:String,@Query("content")content: String): Call<HttpBinResponse>

    @GET("/get")
    fun get(@QueryName name:String): Call<ResponseBody>

    @POST("/post")
    fun post(): Call<ResponseBody>

    @POST("/post")
    @FormUrlEncoded
    fun post(@Field("name") name:String,@Field("content") content:String): Call<ResponseBody>

    @POST("/post")
    @Multipart
    fun post(@Part("va") va:RequestBody,@Part("name\"; filename=\"name.png")name: File, @Part("name2\"; filename=\"name2.png")name2: File): Call<ResponseBody>

    @PATCH("/patch")
    @FormUrlEncoded
    fun patch(@Field("name") name:String,@Field("content") content:String): Call<ResponseBody>

    @PUT("/put")
    @FormUrlEncoded
    fun put(@Field("name") name:String,@Field("content") content:String): Call<ResponseBody>

    @DELETE("/delete")
    fun delete(): Call<ResponseBody>
}
